src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,217 @@
+                  return outputVerticesRef[aindex].getCoords()[effectiveCoordinate] < outputVerticesRef[bindex].getCoords()[effectiveCoordinate];
+                });
+      // Compute the midpoints of the 1D output mesh
+      for (size_t i = 0; i < (outSize - 1); i++) {
+        auto axisPositionCurrent = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        auto axisPositionNext    = output()->vertices()[i + 1].getCoords()[effectiveCoordinate];
+        axisMidpoints(i)         = (axisPositionCurrent + axisPositionNext) / 2;
+      }
+      axisMidpoints(outSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
+
+      std::vector<int> counters(outSize); // counts number of vertices in between midpoints for averaging
+
+      // Identify which vertex (index) of the 3D mesh corresponds to which vertex (index) of the 1D meesh
+      _vertexIndicesCollect.clear();
+      _vertexIndicesCollect.reserve(input()->vertices().size());
+      for (size_t i = 0; i < inSize; i++) {
+        auto vertexCoords = input()->vertices()[i].getCoords()[effectiveCoordinate];
+        int  index        = 0;
+        while (vertexCoords > axisMidpoints(index)) {
+          PRECICE_ASSERT(index + 1 < static_cast<int>(outSize));
+          index++;
+        }
+        _vertexIndicesCollect.push_back(index);
+        counters[index] += 1;
+      }
+      _vertexCounter = std::move(counters);
+    }
+
+  } else {
+    PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+    PRECICE_ASSERT(false, "Not yet implemented");
+    PRECICE_DEBUG("Compute conservative mapping");
+  }
+  _hasComputedMapping = true;
+}
+
+void RadialGeoMultiscaleMapping::clear()
+{
+  PRECICE_TRACE();
+  _vertexIndicesSpread.clear();
+  _vertexIndicesCollect.clear();
+  _hasComputedMapping = false;
+}
+
+void RadialGeoMultiscaleMapping::mapConservative(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+  PRECICE_ASSERT(false, "Not yet implemented");
+  PRECICE_DEBUG("Map conservative");
+}
+
+void RadialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_TRACE();
+
+  const int              inDataDimensions = inData.dataDims;
+  const Eigen::VectorXd &inputValues      = inData.values;
+  Eigen::VectorXd &      outputValues     = outData;
+
+  size_t const inSize  = input()->vertices().size();
+  size_t const outSize = output()->vertices().size();
+
+  PRECICE_ASSERT((inputValues.size() / inDataDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), inDataDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / inDataDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), inDataDimensions, output()->vertices().size());
```

Comment:
So, this means that you force inDataDimensions to be outDataDimensions?

```
PRECICE_ASSERT(!output()->vertices().empty());
auto outDataDimensions = outValues.size() / outSize;

PRECICE_ASSERT(inDataDimensions == outDataDimensions);
```

Reply:
> So, this means that you force inDataDimensions to be outDataDimensions?

Yes, since we don't yet have support for 1D meshes.
However, this was meant as a placeholder to complete the assertion once we can use the right dimensions.

I kept the assertions as-is (also in the axial), but asked for the outDataDimensions. I do agree that these assertions don't make much sense, but I don't feel certain enough to remove them.

In any case, I added the in == out assertion on top of that.